packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { useTree, UseTreeResult, UseTreeOptions } from './useTree';
+
+export interface UseVirtualTreeOptions extends Omit<UseTreeOptions, 'selectedItemIds' | 'defaultSelectedItemIds'> {}
+
+export interface UseVirtualTreeResult extends UseTreeResult {
+  /** ref to be assigned to react-window VariableSizeList/FixedSizeList component */
+  listRef: React.MutableRefObject<any>;
+}
+
+export function useVirtualTree(props: UseVirtualTreeOptions): UseVirtualTreeResult {
+  const baseTree = useTree(props);
+  const {
+    registerItemRef: baseRegisterItemRef,
+    expandSiblings: baseExpandSiblings,
+    getItemById,
+    getItemRef,
+    visibleItemIds,
+  } = baseTree;
+
+  const listRef = React.useRef();
```

Comment:
The easiest way is to use:

```
const listRef = React.useRef<VariableSizeList>();
```

However, it will put a dependency on `react-window` 💣 

```tsx
export interface VariableSizeListRef extends React.Component<any> {
  scrollTo(scrollOffset: number): void;
  scrollToItem(index: number, align?: 'auto' | 'smart' | 'center' | 'end' | 'start'): void;

  resetAfterIndex(index: number, shouldForceUpdate?: boolean): void;
}
```

```diff
-  listRef: React.MutableRefObject<any>;
+  listRef: React.MutableRefObject<VariableSizeListRef>;
```

```diff
-  const listRef = React.useRef();
+  const listRef = React.useRef<VariableSizeListRef>();
```